src/diagnostics/diagnostic.jl

```diff
@@ -49,24 +51,29 @@ Keyword arguments
                              `.=`).
 
 """
-function add_diagnostic_variable!(;
+function add_diagnostic_variable!(
+    requested_diags;
     short_name,
     long_name,
     standard_name = "",
     units,
     comments = "",
     compute!,
 )
-    ALL_DIAGNOSTICS[short_name] = DiagnosticVariable(;
-        short_name,
-        long_name,
-        standard_name,
-        units,
-        comments,
-        compute!,
-    )
+    if isnothing(requested_diags) || short_name in requested_diags
```

Comment:
Instead of using `requested_diags == nothing`, we can use `requested_diags = get_possible_diagnostics(model)` to get the entire set of diagnostics available for a model. Then we can just have:
```suggestion
    if short_name in requested_diags
```
and this will have the same effect of not restricting the DiagnosticVariables that are constructed